app/_src/gateway-operator/guides/autoscaling-workloads/datadog.md

```diff
@@ -0,0 +1,230 @@
+
+## Annotate {{ site.kgo_product_name }} with Datadog checks config
+
+{:.note}
+> **Note:** {{ site.kgo_product_name }} uses [kube-rbac-proxy][kuberbacproxy_github]
+> to secure its endpoints behind an RBAC proxy. That's why we'd going to scrape `kube-rbac-proxy`
+> and not the `manager` container.
+
+
+In order for Datadog to scrape {{ site.kgo_product_name }}'s metrics we need to let it know how to do it.
+This can be done through the following annotation on {{ site.kgo_product_name }}'s Pod:
+
+```yaml
+ad.datadoghq.com/kube-rbac-proxy.checks: |
+  {
+    "openmetrics": {
+      "instances": [
+        {
+          "bearer_token_auth": true,
+          "bearer_token_path": "/var/run/secrets/kubernetes.io/serviceaccount/token",
+          "tls_verify": false,
+          "tls_ignore_warning": true,
+          "prometheus_url": "https://%%host%%:8443/metrics",
+          "namespace": "autoscaling",
+          "metrics": [
+              "kong_upstream_latency_ms_bucket",
+              "kong_upstream_latency_ms_sum",
+              "kong_upstream_latency_ms_count",
+            ],
+          "send_histograms_buckets": true,
+          "send_distribution_buckets": true
+        }
+      ]
+    }
+  }
+```
+
+After applying the above you should see `avg:autoscaling.kong_upstream_latency_ms{service:echo}` metrics in your Datadog Metrics explorer.
+
+[kuberbacproxy_github]: https://github.com/brancz/kube-rbac-proxy
+
+## Use `DatadogMetric` to configure cluster agent to expose external metric
+
+In order to be able to use an external metric in `HorizontalPodAutoscaler` we have
+to instruct Datadog agent to expose one for use.
+
+There are several ways to achieve this but we'll use a Kubernetes native way and
+use [`DatadogMetric` CRD][ddmetricguide]:
+
+```yaml
+echo '
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogMetric
+metadata:
+  name: echo-kong-upstream-latency-avg
+  namespace: default
+spec:
+  query: autoscaling.kong_upstream_latency_ms{service:echo} ' | kubectl apply -f -
+```
+
+When Datadog's agent calculates this metric for you, it will update its status:
+
+```bash
+kubectl get -n default datadogmetric echo-kong-upstream-latency-avg -w
+NAME                             ACTIVE   VALID   VALUE               REFERENCES         UPDATE TIME
+echo-kong-upstream-latency-avg   True     True    24.46194839477539                      38s
+```
+
+If everything works correctly, in a couple of seconds you should be able to get the metric via Kubernetes External Metrics API:
+
+```bash
+kubectl get --raw "/apis/external.metrics.k8s.io/v1beta1/namespaces/default/datadogmetric@default:echo-kong-upstream-latency-avg" | jq
+{
+  "kind": "ExternalMetricValueList",
+  "apiVersion": "external.metrics.k8s.io/v1beta1",
+  "metadata": {},
+  "items": [
+    {
+      "metricName": "datadogmetric@default:echo-kong-upstream-latency-avg",
+      "metricLabels": null,
+      "timestamp": "2024-03-08T18:03:02Z",
+      "value": "24233138021n"
```

Comment:
`n` stands for nano? You can mention that this way Datadog exposes it

Reply:
👍 Added a note about that.